phlib/mapldr.c

```diff
@@ -1052,6 +1052,193 @@ PVOID PhGetLoaderEntryDllBase(
     return baseAddress;
 }
 
+/**
+ * Converts a system DLL init block from the current OS version layout to the latest layout.
+ *
+ * \param[in] Source A buffer returned or copied from LdrSystemDllInitBlock.
+ * \param[out] Destination A buffer for storing a version-independent copy of the structure.
+ */
+VOID PhCaptureSystemDllInitBlock(
+    _In_ PVOID Source,
+    _Out_ PPS_SYSTEM_DLL_INIT_BLOCK Destination
+    )
+{
+    ULONG sourceSize;
+    PS_SYSTEM_DLL_INIT_BLOCK copy = { 0 };
```

Comment:
I think a copy could be avoided here by zeroing the output `Desintation` and just populating that instead of the intermediate `copy`.